packages/std/src/query/distribution.rs

```diff
@@ -22,5 +34,62 @@ pub struct DelegatorWithdrawAddressResponse {
 }
 
 impl_response_constructor!(DelegatorWithdrawAddressResponse, withdraw_address: Addr);
-
 impl QueryResponseType for DelegatorWithdrawAddressResponse {}
+
+/// See <https://github.com/cosmos/cosmos-sdk/blob/c74e2887b0b73e81d48c2f33e6b1020090089ee0/proto/cosmos/distribution/v1beta1/query.proto#L169-L178>
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct DelegationRewardsResponse {
+    pub rewards: Vec<DecCoin>,
+}
+
+impl_response_constructor!(DelegationRewardsResponse, rewards: Vec<DecCoin>);
+impl QueryResponseType for DelegationRewardsResponse {}
+
+/// A coin type with decimal amount.
+/// Modeled after the Cosmos SDK's [DecCoin](https://github.com/cosmos/cosmos-sdk/blob/c74e2887b0b73e81d48c2f33e6b1020090089ee0/proto/cosmos/base/v1beta1/coin.proto#L32-L41) type
```

Comment:
```suggestion
/// Modeled after the Cosmos SDK's [DecCoin] type.
/// However, in contrast to the Cosmos SDK the `amount` string MUST always have a dot at JSON level,
/// see <https://github.com/cosmos/cosmos-sdk/issues/10863>.
/// Also if Cosmos SDK choses to migrate away from fixed point decimals
/// (as shown [here](https://github.com/cosmos/cosmos-sdk/blob/v0.47.4/x/group/internal/math/dec.go#L13-L21 and discussed [here](https://github.com/cosmos/cosmos-sdk/issues/11783)),
/// wasmd needs to truncate the decimal places to 18.
///
/// [DecCoin]: (https://github.com/cosmos/cosmos-sdk/blob/v0.47.4/proto/cosmos/base/v1beta1/coin.proto#L28-L38)
```

Reply:
Also great addition. How is that bug not documented at all on the `LegacyDec` in the sdk o_O

I think truncating to 18 places should be fine if they ever decide to change it.